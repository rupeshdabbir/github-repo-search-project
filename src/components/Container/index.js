import React from 'react';
import {
  StyleSheet,
  css,
} from 'aphrodite/no-important';

type Props = {
  rootStyle?: any,
  children: any,
  noGutter: boolean
};

export default function Container({noGutter, rootStyle, children}: Props) {
  const className = css(
    styles.Container_inner,
    noGutter ? false : styles.Container_padding,
  );

  return (
    <div className={css(styles.Container, rootStyle)}>
      <div className={className}>
        {children}
      </div>
    </div>
  );
}

Container.defaultProps = {
  noGutter: false,
};

const styles = StyleSheet.create({
  Container: {
    margin: 0,
  },

  Container_inner: {
    margin: '0 auto',
    maxWidth: 1020,
  },

  Container_padding: {
    paddingLeft: 10,
    paddingRight: 10,

    '@media (min-width: 1050px)': {
      padding: 0,
    },
  },
});
