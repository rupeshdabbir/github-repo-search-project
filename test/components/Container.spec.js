import React from 'react';
import renderer from 'react-test-renderer';
import {
  StyleSheetTestUtils,
  StyleSheet,
} from 'aphrodite';

import Container from 'components/Container';

describe('Component: Container', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('should render', () => {
    const component = renderer.create(
      <Container>
        <div>inner component</div>
      </Container>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render style overrides', () => {
    const styles = StyleSheet.create({
      test: {color: 'red'},
    });
    const component = renderer.create(
      <Container extraStyle={styles.test}>
        <div>inner component</div>
      </Container>
    );
    expect(component).toMatchSnapshot();
  });

});

