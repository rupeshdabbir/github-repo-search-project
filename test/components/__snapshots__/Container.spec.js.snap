// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Container should render 1`] = `
<div
  className="Container_16pg94n"
>
  <div
    className="Container_inner_w9z6jj-o_O-Container_padding_1cv73u5"
  >
    <div>
      inner component
    </div>
  </div>
</div>
`;

exports[`Component: Container should render style overrides 1`] = `
<div
  className="Container_16pg94n"
>
  <div
    className="Container_inner_w9z6jj-o_O-Container_padding_1cv73u5"
  >
    <div>
      inner component
    </div>
  </div>
</div>
`;
